### PR TITLE
feat: add lint+hint for recipe parsing

### LINKED ISSUE #2140: lint on recipe parseability
The bot and many other tools rely on something being able to parse the recipe YAML. 

Right now there are a few such parsers for v0 recipes in existence. They include

- conda-recipe-manager: https://github.com/conda-incubator/conda-recipe-manager
- conda-forge-tick (the bot): https://github.com/regro/cf-scripts/blob/main/conda_forge_tick/recipe_parser/_parser.py#L486
- (conda-)souschef (used by grayskull): https://github.com/marcelotrevisani/souschef

I propose that we issue a

- lint that fails for conda-forge recipes if they cannot be parsed by any of these tools
- hint per tool if the recipe cannot be parsed by that tool

The new v1 recipes are always parseable so this would not apply to them.

Thoughts @conda-forge/core?

### PR DESCRIPTION
**Pull request by beckermr**
Checklist
* [x] Added a ``news`` entry
* [x] Regenerated schema JSON if schema altered (`python conda_smithy/schema.py`)

This PR is a good step for moving towards v1 recipes. We'll need recipes to be able to be parsed by CRM in order to try automatic conversion. I also included the bot's parser and the one used by grayskull.

closes #2140

### CODE REVIEW

**Comment by beckermr** [comment]
@conda-forge/core this one is ready for review!

File: conda_smithy/linter/lints.py [hunk 1]
```diff
@@ -983,3 +987,100 @@ def sort_osx(versions):
...
+
+
+def lint_recipe_is_parsable(
+    recipe_text: str,
+    lints: List[str],
+    hints: List[str],
+    recipe_version: int = 0,
+):
+    parse_vars = []
+    parser_names = []
```
Comment (jaimergp) [inline conda_smithy/linter/lints.py:999]: I think this should be a `dict` so we don't rely on the implicit pairing of the values.

File: conda_smithy/linter/lints.py [hunk 2]
```diff
@@ -983,3 +987,100 @@ def sort_osx(versions):
...
+        try:
+            from souschef.recipe import Recipe
+        except ImportError:
+            parses_with_souschef = None
+            pass
+        else:
+            with tempfile.TemporaryDirectory() as tmpdir:
+                recipe_file = os.path.join(tmpdir, "meta.yaml")
+                with open(recipe_file, "w") as f:
+                    f.write(recipe_text)
```
Comment (jaimergp) [inline conda_smithy/linter/lints.py:1032]: Souschef modifies the recipe file on parsing? 🤔

**Review by jaimergp — APPROVED** [review]
Awesome! Only had one small comment but it's non-blocking.

File: conda_smithy/linter/lints.py (hunk 2 above)
Comment (beckermr) [reply conda_smithy/linter/lints.py:1032]: Likely not but why take the risk?

File: conda_smithy/linter/lints.py (hunk 1 above)
Comment (beckermr) [reply conda_smithy/linter/lints.py]: done!

**Pull request merged**